docs/tutorials/gallery_adding_components.py

```diff
@@ -0,0 +1,125 @@
+"""
+# Adding Components to Ragna
+
+Ragna has builtin support for several [assistants][ragna.assistants] and 
+[source storages][ragna.source_storages], but there may be cases where you 
+want to use one that is not currently supported.
```

Comment:
I would phrase this a little different. Ragna was built with the goal in mind that you bring your own components. We only provide very basic ones inside the library. For everything else you'll have to implement your own.